lib/elixir/src/elixir_clauses.erl

```diff
@@ -186,6 +186,13 @@ expand_with_else(Meta, Opts, E, HasMatch) ->
   form_error(Meta, ?key(E, file), elixir_expand, {missing_option, 'try', [do]});
 'try'(Meta, [{do, _}], E) ->
   form_error(Meta, ?key(E, file), elixir_expand, {missing_option, 'try', ['catch', 'rescue', 'after', 'else']});
+'try'(Meta, [{do, _}, {else, _}], E) ->
+  Kind =
+    case lists:keyfind(origin, 1, Meta) of
+      {origin, Origin} -> Origin;
+      false -> 'try'
+    end,
```

Comment:
I believe we already do this elsewhere in this file too, maybe we should put this behind a private function? Something like `origin(Meta, 'try')`?

Reply:
@josevalim I want to do this as the next PR, where I fix the other similar cases I mentioned. Would that work for you?